Builder error when subtracting an Option/Result from another Option/Result

**V full version**: V 0.2.4 9355048.940c78b
**OS**: Windows 10 Home v19043 64-bit

**What did you do?**
I forgot to handle the possible error(s) for `Grid.find()` in this minimum viable example:

```v
type Grid = [][]rune

struct Point {
	x int
	y int
}

fn (a Point) - (b Point) Point {
	return Point{a.x - b.x, a.y - b.y}
}

fn (g Grid) dir(a rune, b rune) Point {
	return g.find(a) - g.find(b)
}

fn (g Grid) find(c rune) ?Point {
	for x in 0 .. g.len {
		for y in 0 .. g.len {
			if g[x][y] == c {
				return Point{x, y}
			}
		}
	}
	return error('character not found in grid')
}
```

**What did you expect to see?**
A V compiler error which told me I needed to properly handle the optionals or some such
 
**What did you see instead?**
<details>
<summary>A C builder error.</summary>

```
$  v -cg run point.v
C:/Users/<user>/AppData/Local/Temp/v_0/point.1443555331632405623.tmp.c:460: warning: WINVER redefined
C:/Users/<user>/AppData/Local/Temp/v_0/point.1443555331632405623.tmp.c:6388: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/<user>/AppData/Local/Temp/v_0/point.1443555331632405623.tmp.c:7031: warning: assignment from incompatible pointer type
C:/Users/<user>/AppData/Local/Temp/v_0/point.1443555331632405623.tmp.c:7031: warning: assignment makes pointer from integer without a cast
C:/Users/<user>/AppData/Local/Temp/v_0/point.1443555331632405623.tmp.c:7031: warning: cast between pointer and integer of different size
C:/Users/<user>/AppData/Local/Temp/v_0/point.1443555331632405623.tmp.c:11701: warning: implicit declaration of function '_option_main__Point__minus'
C:/Users/<user>/AppData/Local/Temp/v_0/point.1443555331632405623.tmp.c:11701: error: '{' expected (got ";")
builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```
</details>

----

The problem seems to arise from the interaction of the overloaded operator and the optional(s).

If I move one of or both of the values being subtracted in Grid.dir() out into a separate variable/statement then V gives me a nice helpful error as expected.

```v
fn (g Grid) dir(a rune, b rune) Point {
	a_point := g.find(a)
	return a_point - g.find(b)
}
```
```
point.v:13:14: error: find() returns an option, so it should have either an `or {}` block, or `?` at the end
   11 |
   12 | fn (g Grid) dir(a rune, b rune) Point {
   13 |     a_point := g.find(a)
      |                 ~~~~~~~
   14 |     return a_point - g.find(b)
   15 | }
point.v:14:8: error: unwrapped optional cannot be used in an infix expression
   12 | fn (g Grid) dir(a rune, b rune) Point {
   13 |     a_point := g.find(a)
   14 |     return a_point - g.find(b)
      |           ~~~~~~~~~~~~~~~~~~~
   15 | }
   16 |
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1145,6 +1145,8 @@ pub fn (mut c Checker) infix_expr(mut node ast.InfixExpr) ast.Type {
 	right_is_optional := right_type.has_flag(.optional)
 	if (left_is_optional && !right_is_optional) || (!left_is_optional && right_is_optional) {
 		c.error('unwrapped optional cannot be used in an infix expression', left_right_pos)
+	} else if left_is_optional && right_is_optional {
+		c.error('unwrapped optionals cannot be used in an infix expression', left_right_pos)
 	}
 	// Dual sides check (compatibility check)
 	if !(c.symmetric_check(left_type, right_type) && c.symmetric_check(right_type, left_type))

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -927,9 +927,10 @@ pub fn (mut c Checker) infix_expr(mut node ast.InfixExpr) ast.Type {
 						}
 					}
 				}
-			} else if left_type.has_flag(.optional) && right_type.has_flag(.optional) {
+			} else if left_type.has_flag(.optional) || right_type.has_flag(.optional) {
+				opt_comp_pos := if left_type.has_flag(.optional) { left_pos } else { right_pos }
 				c.error('unwrapped optional cannot be compared in an infix expression',
-					left_right_pos)
+					opt_comp_pos)
 			}
 		}
 		.left_shift {
@@ -1144,7 +1145,8 @@ pub fn (mut c Checker) infix_expr(mut node ast.InfixExpr) ast.Type {
 	left_is_optional := left_type.has_flag(.optional)
 	right_is_optional := right_type.has_flag(.optional)
 	if (left_is_optional && !right_is_optional) || (!left_is_optional && right_is_optional) {
-		c.error('unwrapped optional cannot be used in an infix expression', left_right_pos)
+		opt_infix_pos := if left_is_optional { left_pos } else { right_pos }
+		c.error('unwrapped optional cannot be used in an infix expression', opt_infix_pos)
 	} else if left_is_optional && right_is_optional {
 		c.error('unwrapped optionals cannot be used in an infix expression', left_right_pos)
 	}

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1144,11 +1144,11 @@ pub fn (mut c Checker) infix_expr(mut node ast.InfixExpr) ast.Type {
 	// TODO move this to symmetric_check? Right now it would break `return 0` for `fn()?int `
 	left_is_optional := left_type.has_flag(.optional)
 	right_is_optional := right_type.has_flag(.optional)
-	if (left_is_optional && !right_is_optional) || (!left_is_optional && right_is_optional) {
+	if left_is_optional && right_is_optional {
+		c.error('unwrapped optionals cannot be used in an infix expression', left_right_pos)
+	} else if left_is_optional || right_is_optional {
 		opt_infix_pos := if left_is_optional { left_pos } else { right_pos }
 		c.error('unwrapped optional cannot be used in an infix expression', opt_infix_pos)
-	} else if left_is_optional && right_is_optional {
-		c.error('unwrapped optionals cannot be used in an infix expression', left_right_pos)
 	}
 	// Dual sides check (compatibility check)
 	if !(c.symmetric_check(left_type, right_type) && c.symmetric_check(right_type, left_type))

--- a/vlib/v/checker/tests/infix_compare_optional_err.out
+++ b/vlib/v/checker/tests/infix_compare_optional_err.out
@@ -1,6 +1,6 @@
 vlib/v/checker/tests/infix_compare_optional_err.vv:6:5: error: unwrapped optional cannot be compared in an infix expression
-    4 |
+    4 | 
     5 | fn main(){
     6 |     if foo() > foo() {}
-      |        ~~~~~~~~~~~~~
+      |        ~~~~~
     7 | }

--- a/vlib/v/checker/tests/infix_err.out
+++ b/vlib/v/checker/tests/infix_err.out
@@ -26,11 +26,11 @@ vlib/v/checker/tests/infix_err.vv:11:7: error: `+` cannot be used with `?int`
       |       ^
    12 | _ = int(0) + g() // FIXME not detected
    13 | _ = g() + int(3)
-vlib/v/checker/tests/infix_err.vv:12:5: error: unwrapped optional cannot be used in an infix expression
+vlib/v/checker/tests/infix_err.vv:12:14: error: unwrapped optional cannot be used in an infix expression
    10 | 
    11 | _ = 4 + g()
    12 | _ = int(0) + g() // FIXME not detected
-      |     ~~~~~~~~~~~~
+      |              ~~~
    13 | _ = g() + int(3)
    14 | _ = g() + 3
 vlib/v/checker/tests/infix_err.vv:13:9: error: `+` cannot be used with `?int`

--- a/vlib/v/checker/tests/unwrapped_optional_infix.out
+++ b/vlib/v/checker/tests/unwrapped_optional_infix.out
@@ -2,5 +2,4 @@ vlib/v/checker/tests/unwrapped_optional_infix.vv:5:9: error: unwrapped optional 
     3 | }
     4 | 
     5 | println(test() == "")
-      |         ~~~~~~~~~~~~
-
+      |         ~~~~~~

